node/pkg/watchers/hedera/watcher.go

```diff
@@ -0,0 +1,634 @@
+	go func() {
+		defer close(errC)
+		t := time.NewTicker(5 * time.Second)
+		client := &http.Client{
+			Timeout: time.Second * 5,
+		}
+		// var beginningTS = time.Now().String()
+		var beginningTS = "1234567890.000000400"
+
+		for {
+			<-t.C
+			logString := "api/v1/contracts/" + watcher.contract + "/results/logs?order=asc&timestamp=gt%3A" + beginningTS
+			logger.Info("current logString", zap.String("string", logString))
+			latency := time.Now()
+			// Query and report height and set currentSlotHeight
+			resp, err := client.Get(fmt.Sprintf("%s/%s", watcher.urlRest, logString))
+			if err != nil {
+				logger.Error("query latest logs response error", zap.String("network", networkName), zap.Error(err))
+				continue
+			}
+			logsBody, err := io.ReadAll(resp.Body)
+			if err != nil {
+				logger.Error("query latest logs response read error", zap.String("network", networkName), zap.Error(err))
+				errC <- err
+				resp.Body.Close()
+				continue
+			}
+			resp.Body.Close()
+
+			// Update the prometheus metrics with how long the http request took to the rest api
+			queryLatency.WithLabelValues(networkName, "block_latest").Observe(time.Since(latency).Seconds())
+
+			var events []*common.MessagePublication
+			logJSON := string(logsBody)
+			// logger.Info("logJSON", zap.String("logJSON", logJSON))
+			logs := gjson.Get(logJSON, "logs")
+			logger.Info("after gjson.get", zap.Stringer("logs", logs))
+			logs.ForEach(func(logKey, logValue gjson.Result) bool {
+				logger.Info("YIKES..............", zap.Stringer("topics", gjson.Get(logValue.String(), "topics")))
+				topics := gjson.Get(logValue.String(), "topics")
+				topics.ForEach(func(topicKey, topicValue gjson.Result) bool {
+					if topicValue.String() == TOPIC_LOG_MSG {
+						event, err := LogMessageToEvent(logger, logValue.String())
+						// Check for event being nil
+						if err == nil {
+							events = append(events, event)
+							blockNum := gjson.Get(logValue.String(), "block_number")
+							messagesObserved.WithLabelValues(networkName).Inc()
+							logger.Info("Found True Log Msg", zap.Stringer("block Number", blockNum))
+						}
+						return false // Found topic. Break out of the inner ForEach() loop
+					}
+					return true // continue ForEach() loop
+				})
+				// update timestamp
+				timeStampBase := gjson.Get(logValue.String(), "timestamp")
+				if !timeStampBase.Exists() {
+					logger.Error("Message has no timestamp field")
+					return true // continue ForEach() loop
+				}
+				beginningTS = timeStampBase.String()
+				logger.Info("Updating beginningTS", zap.String("beginningTS", beginningTS))
+				return true // continue ForEach() loop
+			})
+
+			for _, ev := range events {
+				watcher.msgChan <- ev
+				messagesConfirmed.WithLabelValues(networkName).Inc()
+			}
+		}
+	}()
+
+	select {
+	case <-ctx.Done():
+		// err := c.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
```

Comment:
Don't we need to send the close message to the websocket here to get the other thread loop to drop out with a closed web socket?